datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -58,6 +58,8 @@ pub struct ScalarFunctionExpr {
     // and it specifies the effect of an increase or decrease in
     // the corresponding `arg` to the function value.
     monotonicity: Option<FuncMonotonicity>,
+    // Signature of the function
+    signature: Signature,
```

Comment:
It seems like  only one field is ever read. I wonder if it would be better to copy just this field rather than the entire signature (which is both larger with several allocations, but also might be misleading that this signature information was used somehow more in execution plans.

I worry that the signature information might start being referred to in physical planning

So perhaps something like 

```suggestion
    // Does this function need to be invoked with zero arguments ? 
    supports_zero_argument: bool, 
```

```rust
self.signature.type_signature.supports_zero_argument
```

Reply:
Ok